### ISSUE TITLE
Hive not caching metadata when using Amazon Glue metastore

### ISSUE THREAD

**Issue by wjyao0316** [prestodb/presto#16572]
There was a commit to support Glue caching for glue metastore: https://github.com/prestodb/presto/commit/9831c835957e8ca939a5c9e4a4ea6b0c5a96a0bc

I can confirm the presto version 0.245 I use for testing includes this commit. However it still doesn't work. 

My test to reproduce the issue is to run the same query 3 times for the same glue table and I can each query was invoking the Glue GetPartitions API calls.  

The glue hive config:
```
      "hive.metastore": "glue",
      "hive.metastore.glue.get-partition-threads": "50",
      "hive.metastore.glue.max-connections": "50",
      "hive.metastore-cache-ttl": "100m",
      "hive.metastore-refresh-interval": "100m",
```

Can someone help understand the behavior.

**Comment by pettyjamesm** [comment]
I’m assuming you’re using EMR’s version of presto 0.245?

If you are, then that’s because a specific of a decision we made (I currently work on the EMR team responsible for the presto version shipped). We discovered that the default configuration used in combination with caching would very easily result in a huge volume of Glue API requests to refresh cache entries one at a time (the guava cache interface does not support bulk refreshing, only bulk loading) very late in the release cycle, and had to revert the fix because we didn’t have enough time to fix it before the release shipped.

The next EMR release should include a fix that enables glue caching without the risk of the partition caches generating millions of glue API calls from refreshing.
